### DISCUSSION TITLE
PrestaShop 8.1.0 RC 1

### DISCUSSION THREAD

**Discussion by matks** [PrestaShop/PrestaShop#32702] [Announcements]
This is PrestaShop 8.1.0 first Release Candidate.

It includes multiple bug fixes and improvements in addition to all the changes that were already released in [PrestaShop 8.1.0 Beta 1](https://github.com/PrestaShop/PrestaShop/releases/tag/8.1.0-beta.1).

## Changelog

<details>
<summary>Click here to read the changes since beta 1</summary>
<p>

- Back Office:
  - Improvement:
    - #32609: Fix missing AdminSecurity for 8.1.x (by @boherm)
    - #32060: Add admin breadcrumb hook (by @cvng)
    - #32195: Clean product v2 routing (by @jolelievre)
    - #32194: Deprecate code related to old product page (by @jolelievre)
    - #31968: Final UX improvement in PPV2 (by @jolelievre)
    - #31902: Remove cache clear lock file after it is not needed (by @jolelievre)
    - #31874: Create security_test.yml and move default credentials for api in it (by @mflasquin)
    - #31584: Product page v2 : Footer UX improvements (by @jolelievre)
    - #31691: Add admin tabs hook (by @cvng)
    - #31647: Prevent email enumeration (by @MathiasReker)
    - #31560: Duplicate product for multishop final part (by @jolelievre)
    - #31662: Put multiple image feature toggle into the "new feature" block (by @matthieu-rolland)
  - Bug fix:
    - #32509: Fix css for search dropdown in combination generator (81x) (by @zuk3975)
    - #32424: Update BO stock router (by @M0rgan01)
    - #32318: Translation app URL, refresh and selection issues (by @boherm)
    - #32535: Fix: Empty tracking URL for "in transit" email (by @Codencode)
    - #32492: Fix: Can't use LIMIT clause in Sql request manager (by @Codencode)
    - #32465: Fix: Useless redirection in the Customer Service page (by @Codencode)
    - #32199: Set cover on table image when add image on product (by @mflasquin)
    - #32201: Fix undefined shop_id when updating image (by @zuk3975)
    - #32215: Avoid html being escaped in translation in the alert box (by @eternoendless)
    - #32136: Fix reset all default parameters on currency (by @mflasquin)
    - #32202: Add missing attribute in search categories API (by @jolelievre)
    - #32197: [BC Break] Smarty htmlspecialchars (by @0x346e3730)
    - #32116: Fix resetLanguage for Currencies (by @boherm)
    - #32028: Fix module active flag with group & shop contexts (by @boherm)
    - #31411: Display error message when language file is missing (by @Prestaworks)
    - #31390: Ignore attribute_quantity if stock management disabled (by @bibips)
    - #31866: Fix memory leak on suppliers filter (by @mflasquin)
    - #31820: Handle custom lock file during the cache clear and force module actions one by one (by @jolelievre)
    - #30769: Carrier selection improvement in BO product page (by @zuk3975)
    - #31823: Disable legacy_feature_flag for migrated feature page (by @zuk3975)
    - #31674: Add `_legacy_feature_flag` for customer threads routes (by @boherm)
    - #31776: Remove command handler definition that was removed (by @jolelievre)
    - #31772: Fix wordings in BO for keys WS (by @boherm)
    - #31722: Add `force_hide_bulk_actions_btn` in HelperList for AdminCartsController.php (by @boherm)
    - #31512: Implement ShopConstraint into DeleteProductCommand (by @zuk3975)
    - #31508: Fix errors when submitting empty feature or feature values ppv2 (by @zuk3975)
    - #31463: Validate product image size depending on upload quota configuration (by @zuk3975)
    - #31682: Cast `featureId` into int to fix `InvalidFeatureIdException` (by @boherm)
    - #30957: Redirect to product list with warning message instead of raising an exception when we try to access a product which doesn't exist (by @lartist)
    - #31514: Fix sorting attribute do not works (by @mflasquin)
    - #31667: Catch PrestaShopException to validate sql-requests properly in PHP 8.X (by @boherm)
    - #31568: Fix virtual product file constraints (by @zuk3975)
    - #31654: Fix addslashes not working (by @mflasquin)
    - #31296: Update summary quantity on combination list refresh (by @zuk3975)
    - #31603: Fix generating of preload.tpl (by @kpodemski)
    - #31472: Fix "generate for all shops" button visibility when multiShop is off (by @zuk3975)
    - #31330: Fix access error when accessing product creation modal (by @zuk3975)
    - #31496: Handle shop constraint when searching customer for specific price (by @zuk3975)
    - #31337: Fix open new tab when target _blank exist for new product page link (by @mflasquin)
    - #31523: Multistore table dropdown correction (by @JBLach)
    - #31467: PPV2 Fix virtual file validation (by @zuk3975)
    - #31099: Adjust multishop images behavior in dropzone (by @zuk3975)
    - #31355: Fix invisible label on checkbox (by @mflasquin)
    - #31262: Adjust stock movement label when product is virtual (by @zuk3975)
  - Refactoring:
    - #32438: Fix minimum quantity label (by @Hlavtox)
    - #31928: Move namespace from Core/Domain/Product/AttributeGroup to Core/Domain/AttributeGroup (by @JevgenijVisockij)
    - #31885: Final multishop fields that needed to be renamed (by @jolelievre)
    - #31611: Clean ProductImageMultiShopRepository (by @zuk3975)
    - #31855: Refactor product stock update (by @jolelievre)
    - #31657: Clean product & combination command builders (by @zuk3975)
    - #31425: Cleanup ProductMultiShopRepository (by @zuk3975)
    - #31374: Cleanup StockAvailable repositories (by @zuk3975)
    - #31228: Clean up product v2 js structure (by @zuk3975)
- Front Office:
  - Bug fix:
    - #32290: Fix: Setting DNI mandatory after address creation launches an exception during checkout (by @Codencode)
    - #32345: Hide vouchers with no available quantities (by @tleon)
    - #32061: Fix error when change product qty in cart (by @mflasquin)
    - #31620: Fix sitemap url conflict, allow using the hook with fallbacks (by @Hlavtox)
    - #31485: Fix base image format generation, enable it for new image system, simplify (by @Hlavtox)
    - #31310: Fix thumbnail links for categories and stores, add manufacturer and supplier support (by @Hlavtox)
    - #31319: Fix body classes for manufacturer and supplier listings (by @Hlavtox)
    - #31152: Always use id_country from request if provided (by @atomiix)
    - #31148: Change domain to fix translation issue (by @atomiix)
  - Refactoring:
    - #31688: Add avif to htaccess allowed formats (by @Hlavtox)
- Core:
  - Improvement:
    - #32574: Bump classic theme to 2.1.1 (by @matks)
    - #32562: Update Composer dependencies (by @matks)
    - #32379: Fix translation for 8.1.x (by @lartist)
    - #32130: Release 8.0.4 (by @FabienPapet)
    - #32135: Bump version for translation extraction (by @lartist)
    - #32087: Dump modern mail theme with new wordings (by @lartist)
    - #31285: Allow JS action after module import (by @sowbiba)
    - #31981: Fix wordings (by @lartist)
    - #31912: Upgrade prestashop modules (by @matthieu-rolland)
    - #31932: Bump some module versions (by @lartist)
    - #31913: Bump symfony version  (by @matthieu-rolland)
    - #31907: Bump module ps_distributionapiclient (by @matthieu-rolland)
    - #31909: Bump prestashop to 8.0.3 patch version (by @matthieu-rolland)
    - #31811: Deprecate PS1.6 product list colors + formatting (by @SharakPL)
    - #31754: Correct translation domain names to standardize (by @lartist)
    - #31707: Update composer.lock for modules contactform and ps_contactinfo (by @matthieu-rolland)
    - #31692: Corrects some CORE wordings (by @lartist)
    - #31623: Updated PrestaShop Packages (by @github-actions[bot])
    - #31574: Lock theme classic version to 2.0.8 (by @matthieu-rolland)
    - #31575: Add missing licence  (by @matthieu-rolland)
    - #31558: Bump classic theme to 2.1.0 (by @lartist)
    - #31307: Prepare branch for 8.1.0 build (by @lartist)
    - #31339: Add @experimental to new API-related classes (by @atomiix)
  - Bug fix:
    - #32155: Don't attach logo to emails if text only mails is configured 8.1.x (by @cruftex)
    - #32326: [BC Break] Changelog 8.0.4 (by @FabienPapet)
    - #GHSA-8r4m-5p6p-52rp: Fix arbitrary file ready through sql manager (by @mflasquin)
    - GHSA-fh7r-996q-gvcp Fix possible XSS injection through Validate::isCleanHTML method (by @mflasquin)
    - GHSA-p379-cxqh-q822 Fix executeS method making sure it does selection only (by @mflasquin)
    - #32154: Fix domain translation (by @lartist)
    - #31511: Fix exception 'Template ::base.html.twig is not defined' (by @lmeyer1)
    - #31279: Fix errors en customer email field for registration form in FO and BO (by @lartist)
    - #31503: Filter empty values for modulesFromHooks (by @sowbiba)
    - #31137: Disable module's hook before upgrading it (by @atomiix)
    - #31241: Improved session availability test (by @Hlavtox)
- Installer:
  - Improvement:
    - #32536: Update default catalog 8.1.x (by @ps-jarvis)
    - #32147: Update default catalog 8.1.x (by @ps-jarvis)
    - #32027: Add install error message when a parameter is missing (by @eternoendless)
    - #31697: Link to the "data transparency" page inside installer (by @lartist)
  - Bug fix:
    - #32112: Handle fallback in Tools::clearSf2Cache when container is not available (by @jolelievre)
    - #31787: Installer : Fixed link to docs in Installer (by @Progi1984)
    - #31367: Fix a typo in xml hook list (by @thomasnares)
- Web Services:
  - Bug fix:
    - #32044: Fix WS error: add product with empty unit-price (by @boherm)
    - #31023: Backport #30309 : Fix broken result in webservices for 8.0.x (by @boubkerbribri)
- Localization:
  - Improvement:
    - #32522: Wording for 8.1.0 release (by @0x346e3730)
- Tests:
  - Improvement:
    - #32595: Functional Tests : Remove JS support (by @Progi1984)
    - #30574: Functional Tests : GA for checking TypeScript (by @Progi1984)
    - #32310: Functional Tests : API - Authorization Endpoint (by @Progi1984)
    - #32546: Functional Tests : BO - Modules - Alerts tab - Disable / Enable mobile (by @Progi1984)
    - #32431: Functional Tests : WS - Countries : CRUD (by @Progi1984)
    - #32529: Functional Tests : BO - Design - Image Settings - Check stores image format (by @Progi1984)
    - #32511: Functional Tests : BO - Shop Parameters - Search : Fuzzy search (by @Progi1984)
    - #32419: Functional Tests : Hummingbird - FO - Login : Logout from FO (by @Progi1984)
    - #32417: Functional Tests : CLDR - Search a currency by ISO code (by @Progi1984)
    - #32410: Functional Tests : BO - Design - Image Settings - Check category image format (by @Progi1984)
    - #32221: UI Tests : Unskip tests now that #31812 is fixed (by @Progi1984)
    - #32251: Functional Tests : BO - Design - Image Settings - Check product image format (by @Progi1984)
    - #32123: Nightly : Support for HTTPS (by @Progi1984)
    - #31888: Functional Tests : BO - Design - Image Settings - Image Generation on creation (by @Progi1984)
    - #32025: Functional Tests : CLDR - Search a currency by enable/disable (by @Progi1984)
    - #32041: UI Tests : Use HTTPS (by @Progi1984)
    - #32040: UI Tests : Fixed commands (by @Progi1984)
    - #32032: Functional Tests : Move FO tests to classic directory (by @Progi1984)
    - #31881: Functional Tests : Hummingbird : FO - Login - Create an account (by @Progi1984)
    - #31868: Functional Tests : CLDR - Delete a currency (by @Progi1984)
    - #31861: Functional Tests : BO - Design - Image Settings - Regenerate thumbnail (by @Progi1984)
    - #31818: Functional Tests : CLDR - Edit symbol / format currency (by @Progi1984)
    - #31676: Functional Tests : BO - Design - Image Settings - Image Generation options (by @Progi1984)
    - #31808: Functional Tests : API : Authorization Endpoint (by @Progi1984)
    - #31761: Functional Tests : CLDR - Reset symbol / format settings (by @Progi1984)
    - #31669: Functional Tests : API - Resource Endpoint (by @Progi1984)
    - #31687: Functional Tests : CLDR - Enable/Disable a currency (by @Progi1984)
    - #31102: Functional Tests : API - Basic Test (by @Progi1984)
    - #31591: Functional Tests : Splits Tests in small packages (by @Progi1984)
    - #31471: Functional tests : BO - Design - Image Settings - Delete actions (by @Progi1984)
    - #31257: Functional Tests : Migration TS (Remove public attributes targeting a selector) (by @Progi1984)
    - #31292: Functional Tests : Fixed Warning `Warning: fetchStandardFontData: failed to fetch file "FoxitSans.pfb" with "UnknownErrorException: The standard font "baseUrl" parameter must be specified, ensure that the "standardFontDataUrl" API parameter is provided.".` (by @Progi1984)
  - Bug fix:
    - #32607: Functional Tests : Fixed Mocha typescript loader (by @Progi1984)
    - #31947: Nightly : Fixed Tests `functional/FO/08*` (by @Progi1984)
    - #31800: Replace non existent steps with their replacement (by @jolelievre)
    - #31631: Functional Tests : Add the script `test:functional:productV2` (by @Progi1984)
    - #29751: Fixed testsuite with sebastian/comparator release (by @nicosomb)
  - Refactoring:
    - #32510: Functional tests - Add some tests in 'Design > Theme & logo > Pages configuration' (by @nesrineabdmouleh)
    - #32555: Functional tests - Add some tests in 'BO > Merchandise return' page (by @nesrineabdmouleh)
    - #32545: Functional tests - Fix 'BO > Redisplay cart at login' test (by @nesrineabdmouleh)
    - #32557: Functional tests - Add new test 'Advanced parameters > Enable disable debug mode' (by @nesrineabdmouleh)
    - #32498: Functional tests - Add some tests in module manager page (by @nesrineabdmouleh)
    - #32402: Functional tests - Fix random error in 'FO > User account > Get GDPR data in PDF' (by @nesrineabdmouleh)
    - #32314: Functional tests - Add some tests in module manager page (by @nesrineabdmouleh)
    - #32153: Functional tests - Add new test 'BO > Logs > Log by email' (by @nesrineabdmouleh)
    - #32082: Functional tests - Add new test 'CRUD cart rule > Actions >  Apply to specific product' (by @nesrineabdmouleh)
    - #32018: Functional tests - Add some tests in 'BO > Cart rules > Create cart rule > Condition tab' (by @nesrineabdmouleh)
    - #31895: Functional tests - Refacto 'BO > Customer service' (by @nesrineabdmouleh)
    - #31964: Functional tests - Add new test 'Catalog > CRUD cart rule > Product selection' (by @nesrineabdmouleh)
    - #31911: Functional tests - Fix nightly  'FO > Menu and navigation > Sort products' (by @nesrineabdmouleh)
    - #31845: Functional tests - Add some new tests on 'FO > Navigation and display' (by @nesrineabdmouleh)
    - #31762: Functional tests - Add 2 tests 'BO > Cart rule >  Check priority and disabled status' (by @nesrineabdmouleh)
    - #31701: Functional tests - Add new test 'BO > CRUD cart rules > Enable/disable partial use' (by @nesrineabdmouleh)
    - #31718: Fix nightly 8.0.x - 'BO > Orders > Check invoice' (by @nesrineabdmouleh)
    - #31710: Fix nightly 10-03 - 'BO > Orders > View invoice' (by @nesrineabdmouleh)
    - #31702: Fix AfterEach to not fail trying to take a screenshot (ui-test) (by @boubkerbribri)
    - #31671: Functional tests - Delete the test of best sellers block in home page (by @nesrineabdmouleh)
    - #31608: Functional tests - Refacto 'BO > Catalog > CRUD cart rule with & without code' (by @nesrineabdmouleh)
    - #31590: Functional tests - Add new test 'BO > Orders > Shopping carts > Export shopping carts' (by @nesrineabdmouleh)
    - #31573: Functional  tests - Add new test 'BO > CRUD features and values' (by @nesrineabdmouleh)
    - #31555: Functional tests - Refacto 'BO >International > CRUD country' (by @nesrineabdmouleh)
    - #31552: Functional tests - Add new test 'FO > Checkout > Show details' (by @nesrineabdmouleh)
    - #31509: Functional tests - Add new test 'FO > Checkout > Password reminder' (by @nesrineabdmouleh)
    - #31483: Functional Tests : Refactored Feature Flag (by @Progi1984)
    - #31407: Functional tests - Add new test 'Sign in in checkout page' (by @nesrineabdmouleh)
    - #31426: Functional tests - Add new test 'Add to cart by quick view modal' (by @nesrineabdmouleh)
    - #31492: Functional tests - Fix random errors (by @nesrineabdmouleh)
    - #31479: Functional tests - Fix typo error in addOrder page (by @nesrineabdmouleh)
    - #31477: Functional tests - Fix random errors in Nightly (by @nesrineabdmouleh)
    - #31389: Functional tests - Add scripts to run each test less than 30 minutes (by @nesrineabdmouleh)
    - #31357: Functional tests - Add new test 'FO > checkout page > Select carrier' (by @nesrineabdmouleh)
    
</p>
</details>

<hr /><em>This discussion was created from the release <a href='https://github.com/PrestaShop/PrestaShop/releases/tag/8.1.0-rc.1'>PrestaShop 8.1.0 RC 1</a>.</em>

**Comment by matks** [comment]
The release note has been published https://build.prestashop-project.org/news/2023/prestashop-8-1-rc1-release/ (edited)

**Reply by lmeyer1** [reply]
Has been published or has not been published? That is the question! 😃

**Reply by nicosomb** [reply]
The release note has not been published 😆

**Reply by lmeyer1** [reply]
Oh! I got it now!

> The release *note* has been published

**Reply by matks** [reply]
😆 I am so sorry
